remoteappmanager/static/js/home/configurables.js

```diff
@@ -1,85 +1,56 @@
-        '{{#each options}}' +
-           '<option value="{{this}}">{{this}}</option>' +
-        '{{/each}}' +
-        '</div>'
-    );
+    var resolution_conf_tag = 'resolution';
+    var resolution_component = Vue.component(resolution_conf_tag + '-component', {
+        // Your configurable must have a "value" property
+        props: ['value'],
 
-    var ResolutionModel = function () {
-        // Model for the resolution configurable.
-        var self = this;
-        this.resolution = "Window";
-        this.resolution_options = ["Window", "1920x1080", "1280x1024", "1280x800", "1024x768"];
-       
-        self.view = function (index) {
-            // Creates the View to add to the application entry.
-            var widget = $(view_template({
-                options: self.resolution_options,
-                index: index
-            }));
-            
-            widget.find("select").change(function() {
-                if (this.selectedIndex) {
-                    self.resolution = this.options[this.selectedIndex].value;
-                }
-            });
-            
-            return widget;
-        };
-    };
-    
-    ResolutionModel.prototype.tag = "resolution";
-   
-    ResolutionModel.prototype.as_config_dict = function() {
-        // Returns the configuration dict to hand over to the API request.
-        // e.g. 
-        // {
-        //    "resolution" : "1024x768"
-        // }
-        // The returned dictionary must be added to the configurable
-        // parameter under the key given by the tag member. 
-        var resolution = this.resolution;
-        
-        if (resolution === 'Window') {
-            var max_size = utils.max_iframe_size();
-            resolution = max_size[0]+"x"+max_size[1];
+        template:
+            '<div class="form-group">' +
+            '  <label>Resolution</label>' +
+            '  <select class="form-control" v-model="selected_value">' +
+            '    <option v-for="resolution_option in resolution_options">' +
+            '      {{resolution_option}}' +
+            '    </option>' +
+            '  </select>' +
+            '</div>',
+
+        data: function() {
+            return {
+                selected_value: this.value,
+                resolution_options: ['Window', '1920x1080', '1280x1024', '1280x800', '1024x768']
+            };
+        },
+
+        watch: {
+            value: function() {this.selected_value = this.value;}, // model -> view update
+            selected_value: function() {this.$emit('update:value', this.selected_value);} // view -> model update
         }
-        
-        return {
-            "resolution": resolution
-        };
-    };
-   
-    // Define all your configurables here.
-    var configurables = {
-        ResolutionModel: ResolutionModel
+    });
+
+    // Your configurable class must implement tag and value attributes and as_config_dict method
+    var resolution_model = function() {
```

Comment:
which brings me to: do we really need it? I mean, technically, we could
- have only the component.
- have a component method `as_config_dict` that extracts the data from the component.
- have the parent call this method when submit is called, aggregate all responses, and send them as part of the request